BAT/BCEngineMCMC.h

```diff
@@ -1411,13 +1450,29 @@ class BCEngineMCMC
     /** @} */
 
 private:
+
+    // grant friendship to BCModel to access the ThreadLocalStorage
+    friend class BCModel;
```

Comment:
Shouldn't `ThreadLocalStorage` then be in the `protected` section?

Reply:
I assumed you put it in private so that no _user_ classes inheriting from `BCEngineMCMC` could touch it (which makes sense for preserving thread safety). If we made it `protected` couldn't any class derived from `BCModel` (i.e. all user models) access it? As it is now, unless I'm wrong, `BCModel` can access it but classes derived from `BCModel` cannot.